packages/network-controller/src/clients/createInfuraClient.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  createScaffoldMiddleware,
+  JsonRpcMiddleware,
+  mergeMiddleware,
+} from 'json-rpc-engine';
+import {
+  createBlockRefMiddleware,
+  createRetryOnEmptyMiddleware,
+  createBlockCacheMiddleware,
+  createInflightCacheMiddleware,
+  createBlockTrackerInspectorMiddleware,
+  providerFromMiddleware,
+} from 'eth-json-rpc-middleware';
+
+import { createInfuraMiddleware } from '@metamask/eth-json-rpc-infura';
+import { PollingBlockTracker } from 'eth-block-tracker';
+
+import { NetworksChainId, NetworkType } from '@metamask/controller-utils';
+import { CreateClientResult } from './types';
+
+export type InfuraNetworkType =
+  | 'kovan'
+  | 'mainnet'
+  | 'rinkeby'
+  | 'goerli'
+  | 'ropsten';
+
+/**
+ * Create client middleware for infura.
+ *
+ * @param network - the network name.
+ * @param projectId - infura project id.
+ * @returns The network middleware and the block tracker.
+ */
+export default function createInfuraClient(
+  network: InfuraNetworkType,
+  projectId: string,
+): CreateClientResult {
+  const infuraMiddleware = createInfuraMiddleware({
+    network,
+    projectId,
+    maxAttempts: 5,
+    source: 'metamask',
+  });
+  const infuraProvider = providerFromMiddleware(infuraMiddleware);
+  // there is a type mismatch for Provider & SafeEventEmitter.
```

Comment:
This will get fixed soon! I merged a PR in `eth-block-tracker` which aligns this type with the one we get from `eth-json-rpc-middleware`: https://github.com/MetaMask/eth-block-tracker/pull/117

Reply:
This is fixed in `eth-block-tracker` 6.1.0.